packages/algob/src/lib/compile.ts

```diff
@@ -45,7 +46,12 @@ export class CompileOp {
   }
 
   readTealAndHash (filename: string): [string, number] {
-    const content = fs.readFileSync(filename, 'utf8');
+    let content = fs.readFileSync(filename, 'utf8');
+
+    // if ext is .msig teal content should be text above logicSig {refer - /assets/sample-text-asc.msig}
+    if (filename.endsWith(msigExt)) {
+      content = content.split("LogicSig: ")[0];
+    }
     return [content, murmurhash.v3(content)];
```

Comment:
In the hash, we need to hash everything, not only the teal code.

Reply:
updated.